test-data/unit/check-annotated.test

```diff
@@ -127,19 +127,26 @@ x = Annotated[int, Meta()]
 reveal_type(x)  # N: Revealed type is "def () -> builtins.int"
 [builtins fixtures/tuple.pyi]
 
+[case testAnnotatedStringLiteralInFunc]
+from typing_extensions import Annotated
+def f1(a: Annotated[str, "metadata"]):
+    pass
+reveal_type(f1)  # N: Revealed type is "def (a: builtins.str) -> Any"
+def f2(a: Annotated["str", "metadata"]):
+    pass
```

Comment:
Could you reveal_type() this one too?

Reply:
And maybe also add a testcase with a generic function, so we can confirm that a typevar in the first arg to Annotated is still picked up correctly.

Reply:
@JelleZijlstra I added a case. Is it correct?